### ISSUE TITLE
Undesired increase of negative logits using OpenMax

### ISSUE THREAD

**Issue by siebenkopf** [Vastlab/vast#60]
This is a general comment on the OpenMax algorithm that is also present in the paper. When computing the revised logit values (which are called the activations in the paper and in the code), the goal is to decrease activations for samples that are far from the class' MAV. However, when the logit is negative, multiplying with a value smaller than one actually **increases** the logit value, cf: https://github.com/Vastlab/vast/blob/2ad7fa23cb8f0f1acb71d6b6d9aced6a70562e0c/vast/opensetAlgos/heuristic.py#L45

Unfortunately, training a network with Categorical Cross-Entropy Loss on top of SoftMax activations (commonly called the SoftMax loss) does not perform any regulation on the mean value of logits, but only the difference of logits between classes is maximized. For example, in a three-class problem, logits -10, 0, 10  and  -30, -20, -10 will arrive at the exact same SoftMax scores.

There are several ways to circumvent the unintentional increase of negative logits. First, we could re-shift the logits such that the total minimum logit (or the minimum considered logit via alpha) is at 0, but this would end up in reducing large positive logits more. Another option would be to apply a change only when the logit is larger than 0. A third option could be to set the weight to be (2-weight) if the logit is below 0.

Anyway, each of these modifications would change the results and reduce the reproducibility of the paper. One might go with the `run_paper_version` flag to ensure the old, wrong behavior.
